app/code/Magento/Catalog/Model/Product/Image.php

```diff
@@ -226,6 +228,13 @@ public function __construct(
         $this->_assetRepo = $assetRepo;
         $this->_viewFileSystem = $viewFileSystem;
         $this->_scopeConfig = $scopeConfig;
+        if ($viewAssetImageFactory == null) {
+            $this->_viewAssetImageFactory = ObjectManager::getInstance()->get(
+                \Magento\Catalog\Model\View\Asset\ImageFactory::class
+            );
+        } else {
+            $this->_viewAssetImageFactory = $viewAssetImageFactory;
```

Comment:
Please check this out: https://github.com/magento/magento2/blob/develop/app/code/Magento/Backend/Block/System/Account/Edit/Form.php#L60

- new dependency shall be added as last one
- there is no need in `if` statement - just make a one-liner
- always use imported class name like `OptionInterface::class`